compiler/rustc_borrowck/src/diagnostics/region_name.rs

```diff
@@ -331,7 +331,12 @@ impl<'tcx> MirBorrowckCtxt<'_, '_, 'tcx> {
 
                     let closure_kind = match def_ty {
                         DefiningTy::Closure(_, args) => args.as_closure().kind(),
-                        DefiningTy::CoroutineClosure(_, args) => args.as_coroutine_closure().kind(),
+                        DefiningTy::CoroutineClosure(_, args) => {
+                            match args.as_coroutine_closure().kind() {
+                                ty::ClosureKind::Fn => ty::ClosureKind::FnOnce,
```

Comment:
didn't look into this myself, but this seems... quite suboptimal.

Why are we not properly updating the kind for coroutine closures instead so that we don't have to deal with this weird behavior here?